### PR TITLE
[PDSDNREQ-7105]Add settings for Calico to enable Prometheus metrics

### PR DESCRIPTION
**Pull request by dmyar21**
### Description
* Need to enable support for collecting metrics from Calico

Fixes # (issue)
[PDSDNREQ-7105]

### Solution
* Added support for collecting metrics from Calico
* Added new templates responsible for supporting the collection of metrics from Calico
* Added a new section to calico.py , responsible for the enrich parameter responsible for enabling the collection of metrics from Calico
* Added patch p1_reinstall_calico.py

### How to apply
run
```
kubemarine install --tasks="deploy.plugins"
```
or
use patch `p1_reinstall_calico.py`

### Test Cases

**TestCase 1**

Test Configuration:

- Hardware: 
- OS: Ubuntu 20.04, 22.04, Centos 7
- Inventory: 

Steps:

1. Deploy Prometheus on cluster
2. Enabling the parameter and adding services responsible for collecting metrics from Calico

Results:

| Before | After |
| ------ | ------ |
| Not support for collecting metrics  | Support for collecting metrics |

### Checklist
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Integration CI passed
- [ ] Unit tests. If Yes list of new/changed tests with brief description
- [ ] There is no merge conflicts

#### Unit tests
Indicate new or changed unit tests and what they do, if any.

### CODE REVIEW

File: kubemarine/resources/configurations/defaults.yaml [hunk 1]
```diff
@@ -444,6 +444,14 @@ plugins:
             source: templates/plugins/calico-ippool.yaml.j2
             destination: /etc/calico/ippool.yaml
             apply_command: 'calicoctl apply -f /etc/calico/ippool.yaml'
+        - template:
```
[the remark this answers is not in the document]
Comment (dmyar21) [reply kubemarine/resources/configurations/defaults.yaml:447]: not actual

File: kubemarine/plugins/calico.py [hunk 2]
```diff
@@ -322,6 +323,16 @@ def enrich_clusterrole_calico_node(cluster, obj_list):
 
     return obj_list
 
+def enrich_crd_felix_configuration(cluster, obj_list):
+
```
Comment (alexarefev) [inline kubemarine/plugins/calico.py:327]: Please add docstrings

Comment (dmyar21) [reply kubemarine/plugins/calico.py]: add docstring

**Comment by koryaga** [comment]
need to update with main, this change should absent kubemarine/patches/p1_fix_nginx_ingress_k8s_1_25_x.py

File: kubemarine/resources/configurations/defaults.yaml [hunk 3]
```diff
@@ -444,6 +444,14 @@ plugins:
...
+            source: templates/plugins/calico-metrics.yaml.j2
+            destination: /etc/calico/calico-metrics.yaml
+            apply_command: 'kubectl apply -f /etc/calico/calico-metrics.yaml'
+        - template:
+            source: templates/plugins/calico-kube-controllers-metrics.yaml.j2
+            destination: /etc/calico/calico-kube-controllers-metrics.yaml
+            apply_command: 'kubectl apply -f /etc/calico/calico-kube-controllers-metrics.yaml'
```
Comment (ilia1243) [inline kubemarine/resources/configurations/defaults.yaml:454]: Why not just add the resources into original calico manifest during enrichment?

Comment (dmyar21) [reply kubemarine/resources/configurations/defaults.yaml:454]: Good idea, but we have to wait for other changes

**Review by Imadzuma — APPROVED** [review]

**Review by alexarefev — APPROVED** [review]

**Review by ilia1243 — APPROVED** [review]

**Pull request merged**